### PR TITLE
Computed Values

### PR DESCRIPTION
**Pull request by johnjeng**
Submitting for viewing and comments.

### CODE REVIEW

File: src/smc-util/misc.coffee [hunk 1]
```diff
@@ -1785,4 +1785,95 @@ exports.local_storage_length = () ->
...
+#     node1 : []
+#     node2 : ["node1"]
+#     node3 : ["node1", "node2"]
+# Returns
+# object =
+#     top_order : Array<string>
+#     info      : Object
+#         num_sources: number
+# Throws error if cyclic
+# Runs in O(N + E) where N is the number of nodes and E the number of edges
```
Comment (williamstein) [inline src/smc-util/misc.coffee]: Is algorithm taken from some reference?  If so, can you cite it?

File: src/smc-webapp/smc-react.coffee [hunk 2]
```diff
@@ -127,6 +186,47 @@ class Store extends EventEmitter
...
+            if type.is_computed
+                throw "Computed value #{prop_name} in #{store_def.name} was declared but no definition was found."
+            functions[prop_name] = () -> @get(prop_name)
+        else
+            functions[prop_name] = store_def[prop_name]
+            delete store_def[prop_name]
+    delete store_def.stateTypes
+    return functions
+
+# Generates selectors based on functions found in own and other
```
Comment (williamstein) [inline src/smc-webapp/smc-react.coffee]: Period at end of sentence?

**Comment by williamstein** [comment]
Note -- conflicts now due to @haraldschilly's merged PR's.

File: src/smc-webapp/smc-react.coffee [hunk 3]
```diff
@@ -517,4 +619,15 @@ exports.ReactDOM = require('react-dom')
 
 if DEBUG
     smc?.redux = redux  # for convenience in the browser (mainly for debugging)
+    exports._internals =
```
Comment (haraldschilly) [inline src/smc-webapp/smc-react.coffee]: incidentally, I'm also poking around in that smc-react file. right here, I added the line `smc?.react = exports`

**Comment by williamstein** [comment]
@johnjeng -- I made some fixes to misc and misc-test related to my own changes (so fixing some test failures I caused).

I think this PR just fine.  However **it breaks the sort by time/filename toggle**.   

screen shot 2016-11-19 at 9 13 50 am

That seemed like an obvious thing to test when reading your code.  Can you fix this ASAP?  Or should I try to?

**Comment by johnjeng** [comment]
Fixed.

**Pull request merged**